extensions/gitignore/src/gitignore.tsx

```diff
@@ -0,0 +1,127 @@
+                  title={selected ? "Deselect" : "Select"}
+                  icon={selected ? Icon.Circle : Icon.Checkmark}
+                  onAction={() => {
+                    toggleSelection(gitignore);
+                  }}
+                />
+                {CopyToClipboardAction([gitignore])}
+                {PasteAction([gitignore])}
+                {PreviewAction(push, [gitignore])}
+              </ActionPanel>
+            }
+          />
+        );
+      })}
+    </React.Fragment>
+  );
+}
+
+export default function Gitignore() {
+  const { push } = useNavigation();
+
+  const [{ gitignoreFiles, lastUpdated, loading }, selectedIds, toggleSelection, refresh] = useGitignore();
+
+  useEffect(() => {
+    clearSearchBar({ forceScrollToTop: false });
+  }, [selectedIds]);
+
+  const selected = gitignoreFiles.filter((gitignoreFile) => selectedIds.has(gitignoreFile.id));
+  const unselected = gitignoreFiles.filter((gitignoreFile) => !selectedIds.has(gitignoreFile.id));
+
+  return (
+    <List isLoading={loading}>
+      <List.Section>
+        {selected.length > 0 ? (
+          <List.Item
+            icon={Icon.Document}
+            title="Create .gitignore From Selection"
+            actions={
+              <ActionPanel>
+                {CopyToClipboardAction(selected)}
+                {PasteAction(selected)}
+                {PreviewAction(push, selected)}
+              </ActionPanel>
+            }
+          />
+        ) : (
+          !loading && (
+            <List.Item
+              icon={Icon.Download}
+              title="Refresh"
+              subtitle={lastUpdated != null ? `Last updated ${lastUpdated.toLocaleString()}` : undefined}
+              actions={
+                <ActionPanel>
+                  <Action title="Refresh" onAction={refresh} />
+                </ActionPanel>
+              }
+            />
+          )
+        )}
+      </List.Section>
+      {/* Render selected files */}
+      {selected && (
+        <List.Section title="Selected">
+          <GitignoreList gitignoreFiles={selected} selected={true} toggleSelection={toggleSelection} />
+        </List.Section>
+      )}
+      {/* Render unselected files */}
+      {unselected && (
+        <List.Section title="Available">
+          <GitignoreList gitignoreFiles={unselected} selected={false} toggleSelection={toggleSelection} />
+        </List.Section>
+      )}
+    </List>
+  );
+}
+
+function PreviewAction(push: (component: React.ReactNode) => void, selected: GitignoreFile[]) {
+  return (
+    <Action
+      title="Preview"
+      icon={Icon.Eye}
+      shortcut={{ modifiers: ["cmd"], key: "p" }}
+      onAction={() => push(<GitignorePreview gitignoreFiles={selected} />)}
+    />
+  );
+}
+
+function PasteAction(selected: GitignoreFile[]) {
+  return <Action title="Paste to App" icon={Icon.TextDocument} onAction={() => exportPaste(selected)} />;
+}
+
+function CopyToClipboardAction(selected: GitignoreFile[]) {
```

Comment:
Let's make those proper React components. Then you can use them like this in the rest of the code ` <PasteAction selected={[gitignore]} />`.
```suggestion
function PasteAction(props: { selected: GitignoreFile[] }) {
  return <Action title="Paste to App" icon={Icon.TextDocument} onAction={() => exportPaste(props.selected)} />;
}

function CopyToClipboardAction(props: { selected: GitignoreFile[] }) {
```

Reply:
I've made this change in the recent commit. Thanks for spotting this.